constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/SpansAlias.java

```diff
@@ -0,0 +1,31 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.EvaluationEnvironment;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.Segment;
+import gov.nasa.jpl.aerie.constraints.time.Spans;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+
+import java.util.Objects;
+import java.util.Set;
+
+public record SpansAlias(String spansAlias) implements Expression<Spans> {
+
+  @Override
+  public Spans evaluate(final SimulationResults results, final Interval bounds, final EvaluationEnvironment environment) {
+    return environment.spansInstances().get(this.spansAlias);
+  }
+
+  @Override
+  public void extractResources(final Set<String> names) {}
```

Comment:
Why is this method empty?

Reply:
It doesn't depend on any resources. (other methods that do modify the names set)

Reply:
Can Spans in general potentially modify resources? If so, in order to extract them, would I need to call something like `SpanAlias("My Span").evaluate(mySimResults, myEnviornment).extractResources`?

Reply:
I'm not really sure what you're saying here.

> Can Spans in general potentially modify resources?

No, nothing in the constraints package can modify resources. Some operations can transform profiles, but the resources in simulation results are never changed.

> SpanAlias("My Span").evaluate(mySimResults, myEnviornment).extractResources

`extractResources` is a method on the expression interface. Its used in the constraints hasura action to associate the resources that an expression used with the violations it produced.

Reply:
I guess I don't know how familiar you are with constraints evaluation. Let me know if I'm not being very clear and I can give you a crash course on the package if you want.